### ISSUE TITLE
Ex2 validation acc vs epochs

### ISSUE THREAD

**Issue by DrJonnyT** [AI-Fundamentals/DiffPrivNPUserModeling#37]
It was flat with high accuracy in testing-  what if you use a model with lower accuracy?
What if you use a less trained model?
Worry is that the data is too easy to learn so it doesn't need the context data

**Comment by DrJonnyT** [comment]
Blocked until #36 finishes

**Comment by DrJonnyT** [comment]
Unblocked
_okay, that's great, can u take the first best model among each (say the best model for 150 users can be at epoch 25) and compute the validation accuracy for say 5000 users for that model and do it for all the models by choosing the first best and their corresponding accuracy? So in x-axis (the model 150, 300, 600 etc with eps 1), y -axis the 150 user model trained for 5000 users (take the validation data),  ect?_

So a script that goes through each folder, loads eval_acc csv file, finds the epoch with highest accuracy, modifies the eval_ntraj settings dictionary and saves as the relevant json

**Comment by DrJonnyT** [comment]
Need to have mean and stdev so need to add stdev calculation to the evaluation

Then need to rerun evaluation for 5000 users

**Comment by DrJonnyT** [comment]
Validation vs epochs
This is all going again now, ex2 evaluation with 5000 users, batch size of 32, job 5461710

**Comment by DrJonnyT** [comment]
Need to add script to plot ex2 validation mean and quartiles
x axis is users seen, which is the number of training users
y axis is accuracy for best epoch during training. Mean/Q25/Q75...Plot it as percentage rather than from 0 - 1

For which epoch to use- instead of taking the epoch with the highest accuracy, use the training loss and pick the epoch with the lowest loss

Start with 0 users, say the accuracy is 0
eps3 is 'orange' color

18 font size for xy label and legend
Set xticks to be the same as n_users in the data

x label is "behaviors seen"

**Comment by DrJonnyT** [comment]
Running ex2 training and val for 50 users. And also then potentially 100. 
**Can't run eval_ntraj until we know which epoch to use**

**Comment by DrJonnyT** [comment]
Complete
